packages/machine/src/protocol/utils/signature-validator.ts

```diff
@@ -1,26 +1,21 @@
-import { recoverAddress } from "ethers/utils";
+import { recoverAddress, Signature } from "ethers/utils";
 
-import { StateChannel } from "../../models";
-import { ProtocolMessage } from "../../protocol-types-tbd";
-import { Context } from "../../types";
+import { EthereumCommitment } from "../../ethereum/types";
 
 export function validateSignature(
-  message: ProtocolMessage,
-  context: Context,
-  state: StateChannel
+  expectedSigner: string,
```

Comment:
Use Address type

Reply:
I'm actually against using `Address` because although it signals to the developer that this field should be an ethereum address, it creates a false sense of safety. I think instead a better practice would be to always write some kind of decorator for functions that expect addresses and instead get either empty string or strings which are not hex values or strings which aren't exactly 42 characters long.

Reply:
Yeah. This is the catch of Typescript: there's no "real" guarantees of type safety.